### PR TITLE
Use GitHub Actions for CI instead of Travis CI

### PR DESCRIPTION
**Pull request by tnir**
This PR also pins Faraday to use to v1/v0 (#374) for production and capybara to 3.36.0 only for test with Ruby 3.0+ (#373).

The following set of test cases is removed from the current Travis CI-settings in this PR since it is hardly used:

- Ruby 2.6 + Rubygems 2.7
- Ruby 2.7 + Rubygems 2.7

## Actions Demo

https://github.com/tnir/geminabox/actions?query=branch%3Aforce-actions

cf. Diff to trigger at the forked repo: https://github.com/tnir/geminabox/compare/github-actions...force-actions

## Refs.
- Blocked by #359 (#363)
- Blocked by #372 (updating #340)
- #374
- #373

Signed-off-by: Takuya Noguchi <takninnovationresearch@gmail.com>

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      fail-fast: false
+      matrix:
+        ruby-version: [2.3.8, 2.4.6, 2.5.4, 2.6.2, 2.7.2, '3.0.0']
+        gem-version: [2.7.9, 3.2.3]
+        exclude:
+          - ruby-version: 3.0.0
+            gem-version: 2.7.9
+        include:
+          - ruby-version: 2.2.10
+            gem-version: 2.7.9
```
Comment (tnir) [inline .github/workflows/ruby.yml:22]: We cannot 

RubyGems 3.2 tests Ruby 2.5+:
https://github.com/rubygems/rubygems/blob/faa8cf6cccd2ae18eba7191618ff639cc0ddfd7c/.github/workflows/ubuntu-rubygems.yml

RubyGems 2.7 tests Ruby 1.8-2.5:
https://github.com/rubygems/rubygems/blob/c66a90aee091bbdf92188c9fdc3439eb2e1684d1/.travis.yml

cf. https://stdgems.org/rubygems/

**Comment by tnir** [comment]
Example results from Actions: https://github.com/tnir/geminabox/actions/runs/974291811

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-16.04
```
Comment (tnir) [inline .github/workflows/ruby.yml:11]: cf. #359

**Comment by tnir** [comment]
cc @sonots

**Comment by tnir** [comment]
Any maintainers?

**Comment by deivid-rodriguez** [comment]
I missed this PR and created a very similar one: #366. Hopefully any of them is attended!

**Comment by tnir** [comment]
@sonots Thoughts?

File: .github/workflows/ruby.yml (hunk 2 above)
Comment (tnir) [reply .github/workflows/ruby.yml]: #363 resolved this problem! 💟

**Comment by tnir** [comment]
11 months ago when I created this PR, it worked well with Ruby 2.6/2.7/3.0 as seen in https://github.com/tnir/geminabox/actions/runs/974291811 while the latest workflow fails only in Ruby 2.6+ as seen in https://github.com/tnir/geminabox/actions/runs/2420969354.

The difference in Actions environment between two runs is Ubuntu 16.04 (deprecated) and Ubuntu 20.04.

All workflows used for this PR: https://github.com/tnir/geminabox/actions?query=branch%3Aforce-actions

**Comment by deivid-rodriguez** [comment]
Could it be Faraday 2.x? Note that this repo does not commit a `Gemfile.lock` file to source control, so Ruby dependencies are also another difference. Faraday 2.x was released in January so it matches the dates you mentioned.

**Comment by tnir** [comment]
geminabox only supports faraday v1, not v2. Faraday v2 removed some methods: https://github.com/lostisland/faraday/blob/v2.0.0/UPGRADING.md#authentication-helper-methods-in-connection-have-been-removed

**Comment by deivid-rodriguez** [comment]
Then locking to Faraday 1.x should do the trick for this PR, and adding faraday 2.x can be re-ticketed.

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    branches: [ master ]
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby-version: [2.3.8, 2.4.6, 2.5.4, 2.6.2, 2.7.2, '3.0.0']
+        gem-version: [3.2.3]
```
Comment (tnir) [inline .github/workflows/ruby.yml:16]: It looks that we cannot use gem 2.7 on Ruby 2.6/2.7/3.0 and gem 3.x has been bundled with Ruby 2.6 and later.

RubyGems 3.2 tests with Ruby 2.5+:
https://github.com/rubygems/rubygems/blob/faa8cf6cccd2ae18eba7191618ff639cc0ddfd7c/.github/workflows/ubuntu-rubygems.yml

RubyGems 2.7 tests with Ruby 1.8-2.5:
https://github.com/rubygems/rubygems/blob/c66a90aee091bbdf92188c9fdc3439eb2e1684d1/.travis.yml

cf. https://stdgems.org/rubygems/

Comment (deivid-rodriguez) [reply .github/workflows/ruby.yml:16]: I didn't get that, tests on TravisCI have been doing what?

Comment (deivid-rodriguez) [reply .github/workflows/ruby.yml:16]: Oh I see. Yeah, I think it adds little benefit to test those combinations. The reason they were still working is that the feature to disallow downgrades is fairly new ([RubyGems 3.2.15](https://github.com/rubygems/rubygems/blob/master/CHANGELOG.md#3215--2021-03-19)), so the downgrades were still working on those older RubyGems. But I agree to remove these entries!

File: README.md [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # Gem in a Box – Really simple rubygem hosting
-[![Build Status](https://secure.travis-ci.org/geminabox/geminabox.png)](http://travis-ci.org/geminabox/geminabox)
+[![Ruby](https://github.com/tnir/geminabox/actions/workflows/ruby.yml/badge.svg)](https://github.com/geminabox/geminabox/actions/workflows/ruby.yml?query=branch%3Amaster)
```
Comment (tnir) [inline README.md:2]: 😭 Would be address in #376

**Pull request merged**